Switch from Http Client to Http Url Connection

Fleker opened the pull request:
This is required in newer versions of Android as Apache's Http Client is removed.

This will solve issue #34

ahorn commented:
Thank you very much. A few minor questions, which once resolved, we can merge:

1. Could you please use [HTTP_OK](https://docs.oracle.com/javase/7/docs/api/java/net/HttpURLConnection.html#HTTP_OK) instead of the literal 200
2. Is the httpclient Maven dependency still needed in the POM file?
3. Could you please update the README to say what versions of Android you think are reasonably supported.
4. Is setDoInput call really needed? The documentation says the [default](https://docs.oracle.com/javase/6/docs/api/java/net/URLConnection.html#setDoInput(boolean)) should work.

Many thanks again!

Fleker commented:
I have made those modifications. 

The HttpClient dependency should no longer be necessary. I have updated the README to include a comment about compatibility. 

The `setDoInput` method is not needed. I just added that from a previous example, though I suppose it isn't necessary.

ahorn commented:
Hi @Fleker  

   For some time now, I wanted to update the README with some contributor information, and references to projects/products that are using android-rss. If you are willing to share any of this information, please feel free to provide the following, or a subset thereof:

* Your name and/or organization you would like to be mentioned;
* The project name, or product, that is using android-rss.

Many thanks again for your contributions!
